### PR TITLE
Support messageFormat as func into PrettyOptions

### PR DESCRIPTION
**Pull request by gtkatakura**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/pinojs/pino-pretty/blob/28ea34be1245122590a978ea233dd9e38d66b695/test/lib/utils.public.test.js#L119-L129
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/pino/index.d.ts [hunk 1]
```diff
@@ -561,8 +563,23 @@ declare namespace P {
...
+         * const logger = pino({
+         *   prettyPrint: {
+         *     ignore: 'requestId',
+         *     messageFormat: (log, messageKey) => {
+         *       const message = log[messageKey];
+         *       if (log.requestId) return `[${log.requestId}] ${message}`;
+         *       return message;
+         *     },
+         *   },
+         * });
```
Comment (AdamVig) [inline types/pino/index.d.ts:579]: This is example is far more detailed than it needs to be. I think we can simplify it, following [the example in the `pino-pretty` documentation](https://github.com/pinojs/pino-pretty#options).
```suggestion
         * {
         *   messageFormat: (log, messageKey) => {
         *     const message = log[messageKey];
         *     if (log.requestId) return `[${log.requestId}] ${message}`;
         *     return message;
         *   }
         * }
```

File: types/pino/index.d.ts [hunk 2]
```diff
@@ -561,8 +563,23 @@ declare namespace P {
         timestampKey?: string;
         /**
          * Format output of message, e.g. {level} - {pid} will output message: INFO - 1123 Default: `false`.
+         *
+         * @example
+         * ```js
```
Comment (AdamVig) [inline types/pino/index.d.ts:568]:
```suggestion
         * ```typescript
```

File: types/pino/index.d.ts [hunk 3]
```diff
@@ -539,6 +539,8 @@ declare namespace P {
         };
     }
 
+    type MessageFormatFunc = (log: Record<string, any>, messageKey: string, levelLabel: string) => string;
```
Comment (AdamVig) [inline types/pino/index.d.ts:542]:
```suggestion
    type MessageFormatFunc = (log: LogDescriptor, messageKey: string, levelLabel: string) => string;
```

**Review by AdamVig — CHANGES_REQUESTED** [review]
As discussed in #51046, it would be preferable to move these types to a new `pino-pretty` package, but I have no qualms about merging this in the meantime.

**Comment by gtkatakura** [comment]
Thank you for your review @AdamVig , I accepted the commit suggestions and apply a git rebase to squash into the initial commit.

**Review by AdamVig — APPROVED** [review]

**Comment by gtkatakura** [comment]
Ready to merge

**Pull request merged**